### PR TITLE
Fix cuDNN8 in cudnn_deconv_layer.cpp

### PR DESCRIPTION
**Pull request by ultinous-garmine**
Please check and then I'll make a release.

### CODE REVIEW

File: src/caffe/layers/cudnn_deconv_layer.cpp [hunk 1]
```diff
@@ -214,15 +218,18 @@ void CuDNNDeconvolutionLayer<Dtype>::Reshape(
...
+    if(!found_conv_algorithm) CUDNN_CHECK(cudnnStatus_t::CUDNN_STATUS_BAD_PARAM);
+
+    // choose backward algorithm for filter
+    // for better or worse, just a fixed constant due to the missing
+    // cudnnGetConvolutionBackwardFilterAlgorithm in cuDNN version 8.0
+    bwd_filter_algo_[i] = CUDNN_CONVOLUTION_BWD_FILTER_ALGO_1;
+    //twice the amount of the forward search to be save
+    workspace_bwd_filter_sizes_[i] = 2 * workspace_fwd_sizes_[i];
+
+    cudnnGetConvolutionBackwardDataAlgorithmMaxCount(Caffe::cudnn_handle(), &max_alg_count);
```
Comment (ultinous-jlocki) [inline src/caffe/layers/cudnn_deconv_layer.cpp]: CUDNNCHECK

Comment (ultinous-jlocki) [reply src/caffe/layers/cudnn_deconv_layer.cpp]: If we left out in simple conv file, please insert there.

**Review by ultinous-jlocki — CHANGES_REQUESTED** [review]

File: src/caffe/layers/cudnn_deconv_layer.cpp [hunk 2]
```diff
@@ -214,15 +218,18 @@ void CuDNNDeconvolutionLayer<Dtype>::Reshape(
...
-    else{
-      // choose backward algorithm for filter
-      // for better or worse, just a fixed constant due to the missing
-      // cudnnGetConvolutionBackwardFilterAlgorithm in cuDNN version 8.0
-      bwd_filter_algo_[i] = CUDNN_CONVOLUTION_BWD_FILTER_ALGO_0;
-      //twice the amount of the forward search to be save
-      workspace_bwd_filter_sizes_[i] = 2 * workspace_fwd_sizes_[i];
-    }
+
+    if(!found_conv_algorithm) CUDNN_CHECK(cudnnStatus_t::CUDNN_STATUS_BAD_PARAM);
```
Comment (ultinous-nil) [inline src/caffe/layers/cudnn_deconv_layer.cpp:222]: Coding style! Body of `if` must be next line
```c++
if(!found_conv_algorithm)
  CUDNN_CHECK(cudnnStatus_t::CUDNN_STATUS_BAD_PARAM);
```

File: src/caffe/layers/cudnn_deconv_layer.cpp [hunk 3]
```diff
@@ -247,7 +254,7 @@ void CuDNNDeconvolutionLayer<Dtype>::Reshape(
         break;
       }
     }
-    if(!found_conv_algorithm) LOG(ERROR) << "cuDNN did not return a suitable algorithm for deconvolution.";
+    if(!found_conv_algorithm) CUDNN_CHECK(cudnnStatus_t::CUDNN_STATUS_BAD_PARAM);
```
Comment (ultinous-nil) [inline src/caffe/layers/cudnn_deconv_layer.cpp:257]: Same here.

File: src/caffe/layers/cudnn_deconv_layer.cpp (hunk 2 above)
Comment (ultinous-garmine) [reply src/caffe/layers/cudnn_deconv_layer.cpp:222]: fml, I won't make a new release for a whitespace fix.

also this is a caffe fork, where is the coding style doc?

File: src/caffe/layers/cudnn_deconv_layer.cpp (hunk 3 above)
Comment (ultinous-garmine) [reply src/caffe/layers/cudnn_deconv_layer.cpp:257]: fml, I won't make a new release for a whitespace fix.

also this is a caffe fork, where is the coding style doc?

**Pull request merged**